cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceUtils.java

```diff
@@ -144,4 +146,46 @@ public static void updateKafkaAutoRebalanceStatus(KafkaStatus kafkaStatus, Kafka
             }
         }
     }
+
+    /**
+     * Updates the {@code KafkaRebalanceStatus} with a new Warning condition based
+     * on the provided {@link Throwable}. If a Warning condition with the same reason
+     * and message already exists, no update is performed.
+     *
+     * @param status    The {@link KafkaRebalanceStatus} object whose conditions will be updated.
+     * @param exception The {@link Throwable} containing the reason and message for the Warning condition.
+     */
+    public static void addWarningCondition(KafkaRebalanceStatus status, Throwable exception) {
+        List<Condition> conditions = status.getConditions() != null ? new ArrayList<>(status.getConditions()) : new ArrayList<>();
+
+        String causeDescription = exception.getCause() != null
+                ? exception.getCause().toString()
+                : "No cause available";
+        Condition newCondition = StatusUtils.buildWarningCondition(causeDescription, exception.getMessage());
+        Condition oldCondition = conditions.stream()
+                .filter(c -> "Warning".equals(c.getType()))
+                .findFirst()
+                .orElse(null);
+
+        if (oldCondition != null) {
+            // If existing Warning condition has same reason & message, do nothing
```

Comment:
Do we not want to update the timestamp in the condition?

Reply:
This was something we debated when putting the proposal for this feature together. We thought it might to be better know when the error first occurred instead when it last occurred.